### DISCUSSION TITLE
Specify theme for image in markdown within a link

### DISCUSSION THREAD

**Discussion by samatcodeapprove** [community/community#27258] [Other Feature Feedback, Questions, & Ideas]
Very excited to see this new feature!
https://github.blog/changelog/2021-11-24-specify-theme-context-for-images-in-markdown/

However it seems to be limited to top-level image tags. It would be great if we could use this feature for images in Markdown that are also links! The classic use case would be having a clickable project logo at the top of my repo README.

**Works**

Markdown:
```md
![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)
![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)
```

Result:
![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)
![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)

**Does not work**

Markdown:
```md
[![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)](https://codeapprove.com)
[![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)](https://codeapprove.com)
```

Result:
[![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)](https://codeapprove.com)
[![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)](https://codeapprove.com)

**Comment by jfullstackdev** [comment]
![GitHub-Mark-Light](https://user-images.githubusercontent.com/3369400/139447912-e0f43f33-6d9f-45f8-be46-2df5bbc91289.png#gh-dark-mode-only)![GitHub-Mark-Dark](https://user-images.githubusercontent.com/3369400/139448065-39a229ba-4b06-434b-bc67-616e2ed80c8f.png#gh-light-mode-only)

I don't know if I'm getting this correctly, it's even working here at the comments, tried too also at my README,

![image](https://user-images.githubusercontent.com/47092464/181181974-0ab2c2b8-533d-4a1d-880a-d00a6bef07f6.png)

**Reply by samatcodeapprove** [reply]
@jdevstatic can you give me the raw Markdown you're using here? Maybe I just am missing something obvious.

**Reply by jfullstackdev** [reply]
I just pasted the example here,

```
![GitHub-Mark-Light](https://user-images.githubusercontent.com/3369400/139447912-e0f43f33-6d9f-45f8-be46-2df5bbc91289.png#gh-dark-mode-only)![GitHub-Mark-Dark](https://user-images.githubusercontent.com/3369400/139448065-39a229ba-4b06-434b-bc67-616e2ed80c8f.png#gh-light-mode-only)
```

**Reply by samatcodeapprove** [reply]
Ah sorry let me give an example of what I am going for:

**Works**

Markdown:
```md
![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)
![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)
```

Result:
![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)
![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)


**Does not work**

Markdown:
```md
[![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)](https://codeapprove.com)
[![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)](https://codeapprove.com)
```

Result:
[![github-tag-darkbg](https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png#gh-dark-mode-only)](https://codeapprove.com)
[![github-tag-lightbg](https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png#gh-light-mode-only)](https://codeapprove.com)

**Reply by jfullstackdev** [reply]
because the image itself is blank

<https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png>

![image](https://user-images.githubusercontent.com/47092464/181185251-cd511e5f-cfd5-40e3-aac1-84c8c127c9f3.png)

the other one is not

<https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png>

![image](https://user-images.githubusercontent.com/47092464/181185380-aed1be42-555c-4124-a900-a391a6c46257.png)

**Reply by samatcodeapprove** [reply]
@jdevstatic no it's not blank! It has white text so it would look good on a dark background. Which is why the image has `#gh-dark-mode-only` in the URL. However that does not work when the image is within a link!

**Reply by jfullstackdev** [reply]
I'm now seeing it on Dark Mode

![image](https://user-images.githubusercontent.com/47092464/181190315-f22c0529-f1bc-4bbd-82f4-aab37054af81.png)

**Comment by bric3** [comment]
You can now drop this in a markdown file

```html
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png">
  <img src="https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png">
</picture>
```

> Example
> <picture>
>  <source media="(prefers-color-scheme: dark)" srcset="https://user-images.githubusercontent.com/109921904/181183840-6d3aeb14-ac81-44c0-b40c-5683945e8dbd.png">
>   <img src="https://user-images.githubusercontent.com/109921904/181183843-2641886e-5e7c-4536-950a-afddc635fd5e.png">
> </picture>

Announced in https://github.com/community/community/discussions/16910
